### PR TITLE
Feature/2826 create index pattern configuration

### PR DESCRIPTION
**Pull request by pablomarga**
Hi team, this PR add the feature of creating new index pattern after the default one is changed in the settings module.

### CODE REVIEW

**Review by asteriscos — APPROVED** [review]
Tested as the issue described and found a strange behavior. When you first set a new index, it works well. BUT, after the **healthcheck**  if you try to set it back to the original one, it does not change the default index.

Steps to reproduce:

1. Set a **new default index** (a new one)
image
image

2. Check it worked in **management->kibana->indexPatterns**
image

3. Set it back to the original `index-pattern`
image

4. Check it didn't actually work
image

**Review by asteriscos — CHANGES_REQUESTED** [review]
Please check my last message, as I detected a strange behavior after I approved the PR.

File: public/components/settings/configuration/components/bottom-bar.tsx [hunk 1]
```diff
@@ -132,27 +133,41 @@ const reloadToast = () => {
...
 
+const executeHealtCheck = () => {
+  getToasts().add({
+    color: 'warning',
+    title: 'You must execute the health check for the changes to take effect',
+    toastLifeTimeMs: 5000,
+    text:
+    <EuiFlexGroup alignItems = "center" gutterSize="s">
+      <EuiFlexItem grow={false} >
+        <EuiButton onClick={() => window.location.href = '#/health-check'} size="s">Execute health check</EuiButton>
```
Comment (frankeros) [inline public/components/settings/configuration/components/bottom-bar.tsx:144]: you should save the toast id and remove it when the button is clicked

File: public/components/wz-menu/wz-menu.js [hunk 2]
```diff
@@ -129,6 +129,44 @@ class WzMenu extends Component {
...
+        AppState.setCurrentPattern(list[0].id);
+      } else {
+        // Check if the current pattern cookie is valid
+        filtered = list.filter(item =>
+          item.id.includes(AppState.getCurrentPattern())
+        );
+        if (!filtered.length) AppState.setCurrentPattern(list[0].id);
+      }
+
+      const data = filtered
```
Comment (frankeros) [inline public/components/wz-menu/wz-menu.js:152]: `filtered` could be true as an empty array

File: public/services/resolves/get-ip.js [hunk 3]
```diff
@@ -26,6 +26,19 @@ export function getIp(
 ) {
   const deferred = $q.defer();
 
+  const checkWazuhConfig = async (indexPatterns) => {
+    const wazuhConfig = new WazuhConfig();
+    const configuration = wazuhConfig.getConfig();
+    let indexPatternFound = indexPatterns.find((indexPattern) => indexPattern.attributes.title === configuration.pattern);
```
Comment (frankeros) [inline public/services/resolves/get-ip.js:32]: `const` instead of `let`

File: public/react-services/pattern-handler.js [hunk 4]
```diff
@@ -45,11 +45,15 @@ export class PatternHandler {
             );
         }
       }
-      if (!patternList.length) {
+
+      const wazuhConfig = new WazuhConfig();
+      const { pattern } = wazuhConfig.getConfig();
+      let indexPatternFound = patternList.find((indexPattern) => indexPattern.title === pattern);
```
Comment (frankeros) [inline public/react-services/pattern-handler.js:51]: `const`

**Review by frankeros — CHANGES_REQUESTED** [review]

**Comment by Desvelao** [comment]
It's working but this message is a little bit strange saying.
```
No valid index patterns were found, proceeding to create default ....
```
image

**Review by frankeros — APPROVED** [review]
Amazing job! 
Congrats @pablomarga :tada:

**Pull request merged**